src/main/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtils.java

```diff
@@ -45,8 +44,12 @@ public static boolean isHadoopUrl(String path) {
      */
     public static InputStream openFile(String path, PipelineOptions popts) {
         try {
+            if (null==path) throw new NullPointerException("path was null, shouldn't be.");
```

Comment:
So this has been a debate in hellbender before.  We ended up deciding to avoid NPE when possible, and prefer IllegalArgumentException.  Objects.nonNull and Preconditions.checkNonNull both throw NPE, so we add utils.nonNull as a replacement.